File: api/src/main/java/me/devsnox/planetsystem/api/handler/DatabaseHandler.java
package me.devsnox.planetsystem.api.handler;

import me.devsnox.planetsystem.api.database.DatabasePlanet;
import me.devsnox.planetsystem.api.database.DatabasePlayer;

import java.util.UUID;

public interface DatabaseHandler {

    DatabasePlanet getPlanet(final UUID uuid);

    void savePlanet(final DatabasePlanet databasePlanet);

    void create(UUID planet, UUID player);

    DatabasePlayer getPlayer(final UUID uuid);

    void savePlayer(DatabasePlayer databasePlayer);

}

File: api/src/main/kotlin/me/devsnox/planetsystem/api/handler/DatabaseHandler.kt
package me.devsnox.planetsystem.api.handler

import me.devsnox.planetsystem.api.database.DatabasePlanet
import me.devsnox.planetsystem.api.database.DatabasePlayer
import java.util.*

interface DatabaseHandler {

	fun getPlanet(uuid: UUID): DatabasePlanet

	fun savePlanet(databasePlanet: DatabasePlanet)

	fun create(planet: UUID, player: UUID, result: (Boolean) -> Unit)

	fun getPlayer(uuid: UUID): DatabasePlayer

	fun savePlayer(databasePlayer: DatabasePlayer)

}
